x-pack/plugins/ml/common/util/parse_interval.ts

```diff
@@ -60,3 +61,18 @@ export function parseInterval(interval: string, checkValidEsUnit = false): Durat
     return null;
   }
 }
+
+/**
+ * Resolves the longest bucket span from the list and multiply it by 2.
+ * @param bucketSpans Collection of bucket spans
+ */
+export function resolveBucketSpanInSeconds(bucketSpans: string[]): number {
+  return (
+    Math.max(
+      ...bucketSpans
+        .map((b) => parseInterval(b))
+        .filter(isDefined)
+        .map((v) => v.asSeconds())
+    ) * 2
```

Comment:
This multiplication by two is currently very specific to its use by the alerting validator, so I would prefer to remove it from here and added in to the processing to `ml/public/alerting/config_validator.tsx`.

Reply:
This function is also used by `alerting_service`, in the part where we create a time range for querying the anomalies indices. But maybe it makes more sense to double the interval in the query and the config validator.

Reply:
The doubling should be done in the validator and `alerting_service` I agree. The function in `job_utils.ts` should just return the maximum bucket span of the specified job(s).